superset-frontend/src/dashboard/util/getComponentWidthFromDrop.js

```diff
@@ -37,39 +37,49 @@ export default function getComponentWidthFromDrop({
     return component.meta.width;
   }
 
-  const draggingWidth = getDetailedComponentWidth({
+  const {
+    width: draggingWidth,
+    minimumWidth: minDraggingWidth,
+  } = getDetailedComponentWidth({
     component,
     components,
   });
 
-  const destinationWidth = getDetailedComponentWidth({
+  const {
+    width: destinationWidth,
+    occupiedWidth: draggingOccupiedWidth,
+  } = getDetailedComponentWidth({
     id: destination.id,
     components,
   });
 
-  let destinationCapacity =
-    destinationWidth.width - destinationWidth.occupiedWidth;
+  let destinationCapacity = Number(destinationWidth - draggingOccupiedWidth);
 
   if (Number.isNaN(destinationCapacity)) {
-    const grandparentWidth = getDetailedComponentWidth({
+    const {
+      width: grandparentWidth,
+      occupiedWidth: grandparentOccupiedWidth,
+    } = getDetailedComponentWidth({
       id: findParentId({
         childId: destination.id,
         layout: components,
       }),
       components,
     });
 
-    destinationCapacity =
-      grandparentWidth.width - grandparentWidth.occupiedWidth;
+    destinationCapacity = Number(grandparentWidth - grandparentOccupiedWidth);
   }
 
-  if (Number.isNaN(destinationCapacity) || Number.isNaN(draggingWidth.width)) {
-    return draggingWidth.width;
+  if (
+    Number.isNaN(destinationCapacity) ||
+    Number.isNaN(Number(draggingWidth))
+  ) {
+    return draggingWidth;
```

Comment:
@ktmud does this look more readable to you? It keeps the original return value here, to play it safe.
you do have to coerce the value twice this way b/c the value is defined in the code multiple times depending on the conditions.

Reply:
Looking good. I think the long-term solution would be to migrate this and all related files to TypeScript so we can be more confident in changes like this.

Reply:
yeah, agree.